Issue with powerplantsmatching

I'm currently setting the model from scratch and have some misunderstandings with powerplantsmatching. Would be very grateful for any suggestions.

## Checklist

- [X] I am using the current [`main`](https://github.com/pypsa-meets-africa/pypsa-africa/tree/main) branch or the latest [release](https://github.com/pypsa-meets-africa/pypsa-africa/releases). Please indicate.
- [X] I am running on an up-to-date [`pypsa-africa` environment](https://github.com/pypsa-meets-africa/pypsa-africa/blob/main/envs/environment.yaml). Update via `conda env update -f envs/environment.yaml`.

## Describe the Bug

Run of the `build_powerplants` rule leads to an `AssertionError: No matches found` caused by `pm.powerplants(from_url=False, update=True, config=config)`. All settings in the config correspond to the `config.default.yaml` except `countries: ["CD"]` and `retrieve_databundle: false`.

The data files with loaded from zenodo manuallly (there were some troubles during automated loading probably due to some tricks of network connection). But structure of the data directories corresponds to one implied by the model and it doesn't look suspicious...

## Error message

*If applicable, paste any terminal output to help illustrating your problem.*
*In some cases it may also be useful to share your list of installed packages: `conda list`.*

```bash
INFO:snakemake.logging:
INFO:pypsa.io:Imported network base.nc has buses, lines
INFO:powerplantmatching.collection:Create combined dataset for GEO, GPD
INFO:powerplantmatching.matching:Comparing data sources `GEO` and `GPD`
Traceback (most recent call last):
  File "~/Documents/_github_/pypsa-africa/.snakemake/scripts/tmpo7x4285k.build_powerplants.py", line 251, in <module>
    pm.powerplants(from_url=False, update=True, config=config)
  File "~/opt/miniconda3/envs/pypsa-africa/lib/python3.10/site-packages/powerplantmatching/collection.py", line 223, in matched_data
    matched = collect(matching_sources, config=config, **collection_kwargs)
  File "~/opt/miniconda3/envs/pypsa-africa/lib/python3.10/site-packages/powerplantmatching/collection.py", line 97, in collect
    matched = combine_multiple_datasets(dfs, datasets, config=config, **dukeargs)
  File "~/opt/miniconda3/envs/pypsa-africa/lib/python3.10/site-packages/powerplantmatching/matching.py", line 258, in combine_multiple_datasets
    crossmatches = link_multiple_datasets(datasets, labels, config=config, **dukeargs)
  File "~/opt/miniconda3/envs/pypsa-africa/lib/python3.10/site-packages/powerplantmatching/matching.py", line 208, in link_multiple_datasets
    return cross_matches(all_matches, labels=labels)
  File "~/opt/miniconda3/envs/pypsa-africa/lib/python3.10/site-packages/powerplantmatching/matching.py", line 147, in cross_matches
    assert not matches.empty, "No matches found"
AssertionError: No matches found
[Sat May 28 14:33:48 2022]
INFO:snakemake.logging:[Sat May 28 14:33:48 2022]
Error in rule build_powerplants:
    jobid: 0
    output: resources/powerplants.csv, resources/powerplants_osm2pm.csv
    log: logs/build_powerplants.log (check log file(s) for error message)

ERROR:snakemake.logging:Error in rule build_powerplants:
    jobid: 0
    output: resources/powerplants.csv, resources/powerplants_osm2pm.csv
    log: logs/build_powerplants.log (check log file(s) for error message)

RuleException:
CalledProcessError in line 329 of ~/Documents/_github_/pypsa-africa/Snakefile:
Command 'set -euo pipefail;  ~/opt/miniconda3/envs/pypsa-africa/bin/python3.10 ~/Documents/_github_/pypsa-africa/.snakemake/scripts/tmpo7x4285k.build_powerplants.py' returned non-zero exit status 1.
  File "~/Documents/_github_/pypsa-africa/Snakefile", line 329, in __rule_build_powerplants
  File "~/opt/miniconda3/envs/pypsa-africa/lib/python3.10/concurrent/futures/thread.py", line 58, in run
ERROR:snakemake.logging:RuleException:
CalledProcessError in line 329 of ~/Documents/_github_/pypsa-africa/Snakefile:
Command 'set -euo pipefail;  ~/opt/miniconda3/envs/pypsa-africa/bin/python3.10 ~/Documents/_github_/pypsa-africa/.snakemake/scripts/tmpo7x4285k.build_powerplants.py' returned non-zero exit status 1.
  File "~/Documents/_github_/pypsa-africa/Snakefile", line 329, in __rule_build_powerplants
  File "~/opt/miniconda3/envs/pypsa-africa/lib/python3.10/concurrent/futures/thread.py", line 58, in run
Removing output files of failed job build_powerplants since they might be corrupted:
resources/powerplants_osm2pm.csv
WARNING:snakemake.logging:Removing output files of failed job build_powerplants since they might be corrupted:
resources/powerplants_osm2pm.csv
Shutting down, this might take some time.
WARNING:snakemake.logging:Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
ERROR:snakemake.logging:Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2022-05-28T143341.241785.snakemake.log
WARNING:snakemake.logging:Complete log: .snakemake/log/2022-05-28T143341.241785.snakemake.log
```

--- a/doc/release_notes.rst
+++ b/doc/release_notes.rst
@@ -35,17 +35,19 @@ Upcoming Release
 
 * Add new countries and update iso code: `PR #330 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/330>`_
 
-* Fix solar pv slope and add correction factor for wake losses `PR #335 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/350>`_
+* Fix solar pv slope and add correction factor for wake losses: `PR #335 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/350>`_
 
-* Add renewable potential notebook `PR #351 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/351>`_
+* Add renewable potential notebook: `PR #351 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/351>`_
 
-* Make cutout workflow simpler `PR #352 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/352>`_
+* Make cutout workflow simpler: `PR #352 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/352>`_
 
-* Add option to run workflow without pop and gdp raster `PR #353 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/353>`_
+* Add option to run workflow without pop and gdp raster: `PR #353 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/353>`_
 
-* Add latitude_optimal to get optimal solar orientation by default `Commit 1b2466b <https://github.com/pypsa-meets-africa/pypsa-africa/commit/de7d32be8807e4fc42486a60184f45680612fd46>`_
+* Add latitude_optimal to get optimal solar orientation by default: `Commit 1b2466b <https://github.com/pypsa-meets-africa/pypsa-africa/commit/de7d32be8807e4fc42486a60184f45680612fd46>`_
 
-* Harmonize CRSs by options `PR #356 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/356>`_
+* Harmonize CRSs by options: `PR #356 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/356>`_
+
+* Fix powerplantmatching problem for DRC and countries with multi-word name: `PR #359 <https://github.com/pypsa-meets-africa/pypsa-africa/pull/359>`_
 
 
 PyPSA-Africa 0.0.2 (6th April 2022)

--- a/scripts/_helpers.py
+++ b/scripts/_helpers.py
@@ -509,7 +509,7 @@ def three_2_two_digits_country(three_code_country):
     return two_code_country
 
 
-def two_digits_2_name_country(two_code_country):
+def two_digits_2_name_country(two_code_country, nocomma=False, remove_start_words=[]):
     """
     Convert 2-digit country code to full name country:
 
@@ -517,6 +517,12 @@ def two_digits_2_name_country(two_code_country):
     ----------
     two_code_country: str
         2-digit country name
+    nocomma: bool (optional, default False)
+        When true, country names with comma are extended to remove the comma.
+        Example CD -> Congo, The Democratic Republic of -> The Democratic Republic of Congo
+    remove_start_words: list (optional, default empty)
+        When a sentence starts with any of the provided words, the beginning is removed.
+        e.g. The Democratic Republic of Congo -> Democratic Republic of Congo (remove_start_words=["The"])
 
     Returns
     ----------
@@ -527,6 +533,25 @@ def two_digits_2_name_country(two_code_country):
         return f"{two_digits_2_name_country('SN')}-{two_digits_2_name_country('GM')}"
 
     full_name = get_country("name", alpha_2=two_code_country)
+
+    if nocomma:
+        # separate list by delim
+        splits = full_name.split(", ")
+
+        # reverse the order
+        splits.reverse()
+
+        # return the merged string
+        full_name = " ".join(splits)
+
+    # when list is non empty
+    if remove_start_words:
+        # loop over every provided word
+        for word in remove_start_words:
+            # when the full_name starts with the desired word, then remove it
+            if full_name.startswith(word):
+                full_name = full_name.replace(word, "", 1)
+
     return full_name
 
 

--- a/scripts/build_powerplants.py
+++ b/scripts/build_powerplants.py
@@ -82,6 +82,11 @@ from shapely import wkt
 
 logger = logging.getLogger(__name__)
 
+# Auxiliary function to adapt to the ppl format
+two_digits_2_nocomma_country_name = lambda x: two_digits_2_name_country(
+    x, nocomma=True, remove_start_words=["The ", "the "]
+)
+
 
 def convert_osm_to_pm(filepath_ppl_osm, filepath_ppl_pm):
 
@@ -141,7 +146,7 @@ def convert_osm_to_pm(filepath_ppl_osm, filepath_ppl_pm):
             )
         )
         .assign(
-            Country=lambda df: df.Country.map(two_digits_2_name_country),
+            Country=lambda df: df.Country.map(two_digits_2_nocomma_country_name),
             Name=lambda df: df.Name,
             # Name=lambda df: "OSM_"
             # + df.Country.astype(str)
@@ -235,7 +240,11 @@ if __name__ == "__main__":
 
     n = pypsa.Network(snakemake.input.base_network)
     countries_codes = n.buses.country.unique()
-    countries_names = list(map(two_digits_2_name_country, countries_codes))
+    countries_names = list(map(two_digits_2_nocomma_country_name, countries_codes))
+
+    # create code name mapping to be used as inverse function
+    country_mapping = dict(zip(countries_names, countries_codes))
+
     config["target_countries"] = countries_names
 
     if "EXTERNAL_DATABASE" in config:
@@ -255,7 +264,7 @@ if __name__ == "__main__":
                     df.Technology.replace("Steam Turbine", "OCGT").fillna("OCGT"),
                 )
             ),
-            Country=lambda df: df.Country.map(country_name_2_two_digits),
+            Country=lambda df: df.Country.map(lambda x: country_mapping[x]),
         )
     )
 

